app/battle_implement/implementation.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from app.entities.knight import Knight
+from app.entities.stuff import Weapon, ArmourPart, SetArmourScope, Potion
+
+
+def crete_armour(armours: list) -> SetArmourScope | list:
+    if armours:
+        armour_parts = [
+            ArmourPart(
+                armour.get("part"), armour.get("protection")
+            )
+            for armour in armours
+        ]
+        armour_scope = SetArmourScope(armour_parts)
+        return armour_scope
+    return SetArmourScope([])
```

Comment:
Using a default value for the armour_parts argument in the init of a class is better than putting an empty list like this` SetArmourScope([])`. Also, I think it's better to change the name of the class, maybe just ArmourScoupe without Set

Reply:
Thanks, it looks really beater!